clients/src/main/java/org/apache/kafka/common/protocol/ProtoUtils.java

```diff
@@ -40,6 +40,12 @@ public static short latestVersion(int apiKey) {
         return Protocol.CURR_VERSION[apiKey];
     }
 
+    public static short oldestVersion(int apiKey) {
```

Comment:
Yea, this will require a bit of refactoring. Current changes are inline with the way `ProtoUtils` is written. Leaving it as it is for now.